__init__.py

```diff
@@ -431,16 +431,22 @@ def modal(self, context, event):
 
     def on_cancel_or_finish(self, context):
         """Called when this operator is finishing (confirm) or got canceled."""
-
-        # Finish the sound recording process.
-        if self.recording_process:
-            self.recording_process.terminate()
-
         # Unregister from the periodic modal calls.
         if self._timer:
             wm = context.window_manager
             wm.event_timer_remove(self._timer)
 
+        # Finish the sound recording process.
+        if self.recording_process:
+            self.recording_process.terminate()
+            # The maximum amount of time for us to wait for ffmpeg to shutdown in seconds
+            maximum_shutdown_wait_time = 3
+            try:
+                # Wait for ffmpeg to exit until we try to read the saved audio file.
+                self.recording_process.wait(maximum_shutdown_wait_time)
+            except subprocess.TimeoutExpired:
+                log.warning(f"ffmpeg did not gracefully shutdown within {maximum_shutdown_wait_time} seconds.")
```

Comment:
Could you please have code lines have max 80 chars?

Reply:
But the max code length in the black config is 100: https://github.com/britalmeida/push_to_talk/blob/master/pyproject.toml#L19

I'm guessing you mean a 100 instead of 80?